synapse/federation/federation_server.py

```diff
@@ -361,7 +361,7 @@ async def process_pdus_for_room(room_id: str):
                             logger.error(
                                 "Failed to handle PDU %s",
                                 event_id,
-                                exc_info=(f.type, f.value, f.getTracebackObject()),
+                                exc_info=(f.type, f.value, f.getTracebackObject()),  # type: ignore
```

Comment:
mypy things that one of these doesn't match the type signature for `logger.error`, but empirically it works...

> synapse/federation/federation_server.py:364: error: Argument "exc_info" to "error" of "Logger" has incompatible type "Tuple[Optional[Type[BaseException]], Optional[BaseException], Any]"; expected "Union[None, bool, Union[Tuple[type, BaseException, Optional[TracebackType]], Tuple[None, None, None]], BaseException]"  [arg-type]

Reply:
@richvdh do you remember why we can't just use `logger.exception` here?

Reply:
er we probably can just `s/error/exception/`, if that helps, though I don't think it will.

or is your question why we pass in an explicit `exc_info` rather than letting `logging` pull it from `sys.exc_info()`? in which case the answer is that `f.getTracebackObject` has magic to build a more useful stacktrace in the face of async execution.  (It's *possible* that it's better nowadays with proper `async/await`, but I suspect it's not.)

Reply:
Thanks for the background! I think leaving the ignore here is the best way forward at the moment without getting into a rabbit hole of logging.